examples/acoustic/fwi_operators.py

```diff
@@ -137,25 +137,20 @@ def GradientOperator(model, v, grad, rec, u, data, time_order=2, spc_order=6,
                              (u.dt2 -
                               s ** 2 / 12.0 * biharmonicu) * v)
 
-    # Create the stencil by hand instead of calling numpy solve for speed purposes
-    # Simple linear solve of a v(t+dt) + b u(t) + c v(t-dt) = L for v(t-dt)
+    # Derive stencil from symbolic equation
     stencil = 1.0 / (2.0 * m + s * damp) * \
         (4.0 * m * v + (s * damp - 2.0 * m) *
          v.forward + 2.0 * s ** 2 * (laplacian + s**2 / 12.0 * biharmonic))
-    # Add substitutions for spacing (temporal and spatial)
-    subs = {s: dt, h: model.get_spacing()}
-    # Add Gradient-specific updates. The dt2 is currently hacky
-    #  as it has to match the cyclic indices
+    eqn = Eq(v.backward, stencil)
 
-    dse = kwargs.get('dse', 'advanced')
-    dle = kwargs.get('dle', 'advanced')
+    # Add expression for receiver injection
+    ti = v.indices[0]
+    receivers = rec.inject(field=v.indexed[ti - 1, ...], offset=model.nbpml,
+                           expr=rec.indexed[time, ...] * dt * dt / m)
 
```

Comment:
Can offset be a "self" property instead, carrying it around looks error prone as it is a lot easier to forget it at one place and destroy everything, at least as a property, even if missed, everything will still be consistent (no different offset for src and rec for example if mistake)